### PR TITLE
Update k8s/minikube versions

### PR DESCRIPTION
**Pull request by radtriste**
**Description of the change:**

**Motivation for the change:**

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by radtriste** [comment]
Jenkins change has been tested ok: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/custom/job/tradisso/job/kogito-serverless-operator-nightly-update-minikube/

File: .ci/jenkins/scripts/minikube.groovy [hunk 1]
```diff
@@ -1,4 +1,4 @@
-minikubeVersion = env.MINIKUBE_VERSION ?: '1.28.0'
+minikubeVersion = env.MINIKUBE_VERSION ?: '1.30.1'
 minikubeKubernetesVersion = env.KUBERNETES_VERSION ?: '1.23.14'
```
Comment (davidesalerno) [inline .ci/jenkins/scripts/minikube.groovy:2]: Don't we need to update also this to version 1.26.3 ?

**Review by desmax74 — APPROVED** [review]

File: .ci/jenkins/scripts/minikube.groovy (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/scripts/minikube.groovy:2]: oh, thought I did ...

```suggestion
minikubeKubernetesVersion = env.KUBERNETES_VERSION ?: '1.26.3'
```

File: .ci/jenkins/scripts/minikube.groovy [hunk 2]
```diff
@@ -48,12 +48,19 @@ void waitForMinikubeStarted() {
     println 'Wait for Minikube components to be in Running state'
     def minikubeStatus = sh(returnStatus: true, script: '''
         set -x
+        function getKubeSystemPodStatus { kubectl get pods $1 -n kube-system -o json | jq '.items[].status.conditions[]'; }
+        function getKubeSystemPodReadyStatus { echo $(kubectl get pods $1 -n kube-system -o json | jq -r '.items[].status.conditions[] | select(.type == "Ready") | .status'); }
```
Comment (MarianMacik) [inline .ci/jenkins/scripts/minikube.groovy:52]: Just an idea for future. What if we had these functions in a separate script which can be sourced?

**Review by MarianMacik — APPROVED** [review]

**Review by davidesalerno — APPROVED** [review]

File: hack/kube-utils.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+getKubeSystemPodStatusConditions() { 
+  kubectl get pods $1 -n kube-system -o json | jq '.items[].status.conditions[]'
+}
+
+getKubeSystemPodReadyStatus() { 
+  echo $(kubectl get pods $1 -n kube-system -o json | jq -r '.items[].status.conditions[] | select(.type == "Ready") | .status') 
+}
+
+export -f getKubeSystemPodStatusConditions
+export -f getKubeSystemPodReadyStatus
```
Comment (MarianMacik) [inline hack/kube-utils.sh:25]: FYI export is not needed if you are sourcing. Export will just make it available in the child scripts, if there are any. I mean, it is not really necessary here. However, this way we could use them in some scripts called from the minikube.groovy script, for example.

**Review by MarianMacik — APPROVED** [review]

**Review by spolti — APPROVED** [review]

**Comment by MarianMacik** [comment]
@radtriste sorry, I didn't notice you call timeout bash... my apologies

**Comment by radtriste** [comment]
@MarianMacik please have a look to that last version :)

**Review by MarianMacik — APPROVED** [review]
Nice!

**Pull request merged**